File: src/main/java/org/droidmate/analyzer/IAppUnderTest.java
package org.droidmate.analyzer;

import org.droidmate.analyzer.api.IApi;
import org.droidmate.analyzer.exploration.IExplorationStrategy;
import org.droidmate.analyzer.exploration.IScenario;

import java.nio.file.Path;
import java.util.List;


/**
 * Application under evaluation
 */
public interface IAppUnderTest {
    Path getApkFile();

    Path getDir();

    IScenario getInitialExpl();

    int getCurrExplDepth();

    List<IScenario> getScenarios(int depth);

    String getPackageName();

    List<IApi> getInitialMonitoredApiList();

    void explore(IExplorationStrategy strategy);

    List<IScenario> getScenarios();

    List<IScenario> getSuccessfulScenarios();

    List<IScenario> getFailScenarios();
}

File: src/main/kotlin/org/droidmate/analyzer/IAppUnderTest.kt
package org.droidmate.analyzer

import org.droidmate.analyzer.api.IApi
import org.droidmate.analyzer.exploration.IExplorationStrategy
import org.droidmate.analyzer.exploration.IScenario

import java.nio.file.Path


/**
 * Application under evaluation
 */
interface IAppUnderTest {
    val apkFile: Path

    val dir: Path

    val initialExpl: IScenario?

    val currExplDepth: Int

    fun getScenariosDepth(depth: Int): List<IScenario>

    val packageName: String

    val initialMonitoredApiList: List<IApi>

    fun explore(strategy: IExplorationStrategy)

    val scenarios: MutableList<IScenario>

    val successfulScenarios: List<IScenario>

    val failScenarios: List<IScenario>
}
